wallet/lib/pages/home/collection_screen/collection_view_model.dart

```diff
@@ -118,9 +128,63 @@ class CollectionViewModel extends ChangeNotifier {
     } on Exception catch (_) {}
   }
 
+  Future loadFavoritesData()async{
+    final response = await repository.getAllFavorites();
+
+    if(response.isRight()){
+      final List<FavoritesModel> favModels = response.getOrElse(() => []);
+      favorites.clear();
+      for(final FavoritesModel favoritesModel in favModels){
+        if(favoritesModel.type == NftType.TYPE_ITEM.name){
+          final item = await repository.getItem(cookBookId: favoritesModel.cookbookId,itemId: favoritesModel.id);
+          if(item.isRight()){
+            favorites.add(await NFT.fromItem(item.toOption().toNullable()!));
+          }
+        }else{
+          final item = await repository.getRecipe(cookBookId: favoritesModel.cookbookId,recipeId: favoritesModel.id);
+          if(item.isRight()){
+            favorites.add(NFT.fromRecipe(item.toOption().toNullable()!));
+          }
+        }
+      }
+    }
+  }
+
+  void shouldShowOwnerViewOrPurchaseViewForNFT({required NFT asset,required BuildContext context}) {
+    if (asset.type == NftType.TYPE_RECIPE) {
+      onRecipeClicked(asset);
+    } else {
+      Navigator.of(context).pushNamed(RouteUtil.ROUTE_OWNER_VIEW, arguments: asset).then((_) => {walletsStore.setStateUpdatedFlag(flag: true)});
```

Comment:
Kindly remove this part `then((_) => {walletsStore.setStateUpdatedFlag(flag: true)});`